docs/designers-developers/developers/tutorials/javascript/js-build-setup.md

```diff
@@ -0,0 +1,228 @@
+# JavaScript Build Setup
+
+This page covers how to setup your development environment to use the ESNext syntax. This documentation covers development for your plugin to work with Gutenberg, to setup a development environment to work directly on Gutenberg, see the [CONTRIBUTING.md](https://github.com/WordPress/gutenberg/blob/master/CONTRIBUTING.md) documentation.
```

Comment:
How about explaining here what `ESNext` is? Something along the lines of:

```suggestion
This page covers how to set up your development environment to use the ESNext syntax. We call ESNext to JavaScript code written using features that are only available in a specification greater than ECMAScript 5 (ES5 for short) or that includes custom syntax such as [JSX](https://reactjs.org/docs/introducing-jsx.html).

This documentation covers development for your plugin to work with Gutenberg, to setup a development environment to work directly on Gutenberg, see the [CONTRIBUTING.md](https://github.com/WordPress/gutenberg/blob/master/CONTRIBUTING.md) documentation.
```

Reply:
Instead of trying to explain here, would it be more helpful to link to a resource on what ESNext is?

Reply:
That may be a good option as well, I just don't know any that I like. Do you have a particular one in mind? As long as it's not assumed the reader already knows what ESNext is, I'm fine.

Reply:
No I didn't have a particular one in mind but maybe this would be sufficient: https://en.wikipedia.org/wiki/ECMAScript#ES.Next?  Or maybe quote from there and link to it as the source?

Reply:
Ah, I see. We use ESNext through the documentation with a different meaning, though (ES>=6 plus JSX).

Reply:
Fro reference this flows in the JavaScript tutorial with the definitions defined a page or two earlier. I update that page with a link here, but can also include a link back for reference if people land on this page prior to visiting the other.

Page is here: https://wordpress.org/gutenberg/handbook/designers-developers/developers/tutorials/javascript/versions-and-building/

Reply:
I think any time we can cross-link, the better.

Maybe we can go even further with expanding the glossary?